trinity/components/builtin/metrics/system_metrics_collector.py

```
@@ -78,6 +87,44 @@ def read_network_stats() -> NetworkStats:
     )
 
 
+@to_tuple
+def get_all_python_processes() -> Iterator[psutil.Process]:
+    for p in psutil.process_iter():
+        try:
+            p.cmdline()
+        except psutil.AccessDenied:
+            continue
+        except psutil.ZombieProcess:
+            continue
+        if 'python' in p.name():
+            yield p
+
+
+def get_main_trinity_process() -> Optional[psutil.Process]:
+    python_processes = get_all_python_processes()
+    for p in python_processes:
+        if 'trinity' in p.cmdline():
+            return p
+    return None
```

Returning `None` will summon Piper haha. Joke aside, we try to avoid returning `None` and Piper has left [some words about it in our tactical manual](https://github.com/ethereum/snake-charmers-tactical-manual/blob/master/antipatterns.md#return-value-checking).

I think this should raise an exception.

---

I laughed when I read this.